game-core/src/main/java/games/strategy/triplea/ui/PoliticalStateOverview.java

```diff
@@ -157,17 +157,16 @@ private JPanel getRelationshipLabel(final PlayerId player1, final PlayerId playe
         data.releaseReadLock();
       }
     }
-    final JComponent relationshipLabel = getRelationshipComponent(player1, player2, relType);
-    final JPanel relationshipLabelPanel = new JPanel();
-    relationshipLabelPanel.add(relationshipLabel);
-    relationshipLabelPanel.setBackground(getRelationshipTypeColor(relType));
-    return relationshipLabelPanel;
+    return getRelationshipComponent(player1, player2, relType, getRelationshipTypeColor(relType));
   }
 
-  private JComponent getRelationshipComponent(
-      final PlayerId player1, final PlayerId player2, final RelationshipType relType) {
+  protected JPanel getRelationshipComponent(
+      final PlayerId player1,
+      final PlayerId player2,
+      final RelationshipType relType,
+      final Color relColor) {
```

Comment:
Is this parameter nulllable? The color is computed from the relationship type, so we probably can avoid the null parameter by moving 'getRelationshipTypeColor' to be within this method. Considering as well the parameter is only used when 'editable' is false, that might be a good move. (Going by principle that ideally all method parameters are always used; which seems to be a problem in this method already, but perhaps better to not exacerbate it)

Going that route, one could have:
```
JPanel panel = new JPanelBuilder().add(new JLabel(relType.getName()).build();
Optional.ofNullable(getRelationshipTypeColor(relType)).ifPressent(panel::setBackgroundColor);
return panel;
```